packages/perseus/src/util/extract-perseus-data.ts

```diff
@@ -269,14 +291,25 @@ function injectWidgets(
         switch (widget.type) {
             case "radio":
                 // Replace radio with the radio options
-                // '[[☃ Radio 1]]' -> 'option 1\noption 2\noption 3'
+                // '[[☃ Radio 1]]' -> 'choice 1\nchoice 2\nchoice 3'
+                // or if the current widget state is available with proper order,
+                // '[[☃ Radio 1]]' ->
+                //   'Option A: choice 1\nOption B: choice 2\nOption C: choice 3'
                 const radio = widget;
+                const radioProps = widgetProps?.[widgetID] as
```

Comment:
I realize that this has already been reviewed and landed, but I'd like to socialize a different approach. 

The radio widget, like all others, supports a `transform` function. This function is the one that does all the heavy lifting to morph the radio widget options into the set that the radio's React component receives and renders. 

So, it would probably mean for more stable less brittle code if we used that transform on the given widget options and then took the `.choices` in the result to derive the order. The only extra bit you'd need is the `problemNum` as that is the randomization seed used. 

https://github.com/Khan/perseus/blob/20a23a3976a70378e86c8871e47d5497ddc9d51e/packages/perseus/src/widgets/radio.ts#L98

Maybe in the future we can refactor to do this.